mongomock/aggregate.py

```diff
@@ -358,6 +358,37 @@ def _handle_date_operator(self, operator, values):
             return out_value.second
         if operator == '$millisecond':
             return int(out_value.microsecond / 1000)
+        if operator == '$dateToString':
+            if not isinstance(values, dict):
+                raise OperationFailure(
+                    '$dateToString operator must correspond a dict'
+                    'that has "format" and "date" field.'
+                )
+            if 'format' not in values.keys() or 'date' not in values.keys():
```

Comment:
Let's move that above:
```py
if not isinstance(values, dict) or not {'format', 'date'} <= set(values.keys()):
```

Reply:
Actually `set(values)` instead of `set(values.keys())`

Reply:
Done